specs/phase1/fraud-proofs.md

```diff
@@ -29,42 +32,259 @@ This document describes the shard transition function and fraud proofs as part o
 
 ## Fraud proofs
 
+### Shard state transition function
+
+```python
+def shard_state_transition(beacon_state: BeaconState,
+                           shard_state: ShardState,
+                           block: ShardBlock) -> None:
+    # Update shard state
+    prev_gasprice = shard_state.gasprice
+    if len(block.body) == 0:
+        latest_block_root = shard_state.latest_block_root
+    else:
+        latest_block_root = hash_tree_root(block)
+
+    shard_state.transition_digest = compute_shard_transition_digest(
+        beacon_state,
+        shard_state,
+        block.beacon_parent_root,
+        block.body,
+    )
+    shard_state.gasprice = compute_updated_gasprice(prev_gasprice, len(block.body))
+    shard_state.slot = block.slot
+    shard_state.latest_block_root = latest_block_root
+```
+
+```python
+def get_post_shard_state(beacon_state: BeaconState,
+                         shard_state: ShardState,
+                         block: ShardBlock) -> ShardState:
+    """
+    A pure function that returns a new post ShardState instead of modifying the given `shard_state`.
+    """
+    post_state = shard_state.copy()
+    shard_state_transition(beacon_state, post_state, block)
+    return post_state
+```
+
+```python
+def compute_shard_transition_digest(beacon_state: BeaconState,
+                                    shard_state: ShardState,
+                                    beacon_parent_root: Root,
+                                    shard_body_root: Root) -> Bytes32:
+    # TODO: use SSZ hash tree root
+    return hash(
+        hash_tree_root(shard_state) + beacon_parent_root + shard_body_root
+    )
+```
+
+```python
+def verify_shard_block_message(beacon_state: BeaconState,
```

Comment:
might be best to move `verify_shard_block_message` and `verify_shard_block_signature` under helpers since they are only used by helper methods